### PR TITLE
Implement implicit form submission from input elements

### PR DESCRIPTION
**Pull request by timbertson**
With submission-blocking rules from https://html.spec.whatwg.org/multipage/forms.html#implicit-submission.

I wasn't sure if/where it would be relevant to add new tests (I couldn't find any existing tests in wpt to enable), so just let me know if this needs tests.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/5865)

### CODE REVIEW

File: components/script/dom/htmlinputelement.rs [hunk 1]
```diff
@@ -812,18 +821,55 @@ impl<'a> Activatable for JSRef<'a, HTMLInputElement> {
...
+            submit_button = node.query_selector_iter("input[type=submit]".to_owned()).unwrap()
                 .filter_map(HTMLInputElementCast::to_temporary)
                 .map(|t| t.root())
-                .find(|r| r.r().form_owner() == owner)
-                .map(|s| s.r().synthetic_click_activation(ctrlKey, shiftKey, altKey, metaKey));
+                .find(|r| r.r().form_owner() == owner);
+        }
+        match submit_button {
+            Some(button) => {
+                button.r().synthetic_click_activation(ctrlKey, shiftKey, altKey, metaKey)
```
Comment (Manishearth) [inline components/script/dom/htmlinputelement.rs]: Check `is_instance_activatable()` before calling this

>  then doing so for a form whose default button has a defined activation behaviour must cause the user agent to run synthetic click activation steps on that default button.

**Comment by Manishearth** [comment]
looks good to me with one issue

**Comment by gfxmonk** [comment]
Thanks, I've fixed that now.

**Comment by Manishearth** [comment]
Reviewed files:
- components/script/dom/htmlinputelement.rs @ r2
- ports/glutin/window.rs @ r1

---





Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/5865)

**Comment by Manishearth** [comment]
Thanks! Please squash the two commits and we'll be good to go!

**Comment by gfxmonk** [comment]
Squashed.

**Comment by jdm** [comment]
@bors-servo: r=Manishearth

Thanks @gfxmonk!

**Pull request merged**